databricks/koalas/frame.py

```diff
@@ -5447,10 +5468,13 @@ def pivot(self, index=None, columns=None, values=None):
             index = [index]
         else:
             df = self.copy()
-            df["__DUMMY__"] = F.monotonically_increasing_id()
-            df.set_index("__DUMMY__", append=True, inplace=True)
-            df.reset_index(level=range(len(df._internal.index_map) - 1), inplace=True)
-            index = df._internal.column_labels[: len(df._internal.index_map)]
+            num_index_col = len(df._internal.index_columns)
+            df.reset_index(inplace=True)
```

Comment:
This causes the default index attachment.
Adding `__DUMMY__` column in the original one is just try to avoid the default index attachment.
Maybe we can just do:

```diff
             df = self.copy()
+            num_index_col = len(df._internal.index_columns)
             df["__DUMMY__"] = F.monotonically_increasing_id()
-            df.set_index("__DUMMY__", append=True, inplace=True)
+            df.set_index(df.columns[-1], append=True, inplace=True)
             df.reset_index(level=range(len(df._internal.index_map) - 1), inplace=True)
-            index = df._internal.column_labels[: len(df._internal.index_map)]
+            index = df._internal.column_labels[:num_index_col]
```

Reply:
Thanks, I addressed!